src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1793,6 +1793,30 @@ export function registerTerminalActions() {
 			accessor.get(ITerminalService).activeInstance?.selectAll();
 		}
 	});
+	registerAction2(class extends Action2 {
+		constructor() {
+			super({
+				id: TerminalCommandId.selectAllAccessibleBuffer,
+				title: { value: localize('workbench.action.terminal.selectAllAccessibleBuffer', "Select All Accessible Buffer"), original: 'Select All Accessible Buffer' },
+				category,
+				precondition: ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated),
+				keybinding: [{
+					// Don't use ctrl+a by default as that would override the common go to start
+					// of prompt shell binding
+					primary: 0,
+					// Technically this doesn't need to be here as it will fall back to this
+					// behavior anyway when handed to xterm.js, having this handled by VS Code
+					// makes it easier for users to see how it works though.
+					mac: { primary: KeyMod.CtrlCmd | KeyCode.KeyA },
+					weight: KeybindingWeight.WorkbenchContrib,
+					when: TerminalContextKeys.accessibleBufferFocus
```

Comment:
there's no `when` clause for the editor select all command and the select all in the terminal as it exists goes right to the text area instead of selecting the contents of the accessible buffer. though maybe that's what we want 🤔 ?

Reply:
so this PR fixes the issue where the accessible buffer is focused and select all is used

Reply:
maybe we shouldn't support selecting all in the accessible buffer anyway though, since it's possible to do in the terminal

Reply:
Oh I see, I was expecting `terminalFocus` to be true.